### PR TITLE
{Rest} az rest: Add examples to --url-parameters

### PR DESCRIPTION
**Pull request by zhoxing-ms**
Issue: #12977

**Description**  

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/util/_help.py [hunk 1]
```diff
@@ -31,6 +31,9 @@
   - name: Create a public IP address from body.json file
     text: >
         az rest --method put --uri https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/publicIPAddresses/{publicIpAddressName}?api-version=2019-09-01 --body @body.json
+  - name: List the top three resources
+    text: >
+        az rest --method get --uri https://management.azure.com/subscriptions/{subscriptionId}/resources?api-version=2019-07-01 --uri-parameters $top=3
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/util/_help.py:36]: - `$` will cause problem in Linux bash as it is interpreted as a variable. 
- We can provide another example to use inline parametera in the URL.

File: src/azure-cli/azure/cli/command_modules/util/_params.py [hunk 2]
```diff
@@ -21,7 +21,8 @@ def load_arguments(self, _):
         c.argument('headers', nargs='+',
                    help="Space-separated headers in KEY=VALUE format or JSON string. Use @{file} to load from a file")
         c.argument('uri_parameters', options_list=['--uri-parameters', '--url-parameters'], nargs='+',
-                   help='Space-separated queries in KEY=VALUE format or JSON string. Use @{file} to load from a file')
+                   help='Query parameters in the URI. Space-separated queries in KEY=VALUE format or JSON string. '
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/util/_params.py:24]:
```suggestion
                   help='Query parameters in the URL. Space-separated queries in KEY=VALUE format or JSON string. '
```

File: src/azure-cli/azure/cli/command_modules/util/_help.py (hunk 1 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/util/_help.py:36]: Yes, `$` works fine in `CMD`, and there will be problems in `Bash` and `PowerShell` that are considered special characters. 
But the escape characters of these environments are different(`bash` is \\ ,`PowerShell` is \` ,`CMD` is ^ ), so what environment is our example mainly based on? Is it `Bash`?

Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/util/_help.py:36]: We have no standard regarding this. Let's say "In Bash, escape `$` as `\$` and in PowerShell, escape `$` as `` `$ ``".

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/util/_help.py:36]: OK~ So I should specify how to escape `$` in different environments in `- name`, right?

**Comment by yonzhan** [comment]
add to S171

File: src/azure-cli/azure/cli/command_modules/util/_params.py [hunk 3]
```diff
@@ -21,7 +21,8 @@ def load_arguments(self, _):
         c.argument('headers', nargs='+',
                    help="Space-separated headers in KEY=VALUE format or JSON string. Use @{file} to load from a file")
         c.argument('uri_parameters', options_list=['--uri-parameters', '--url-parameters'], nargs='+',
-                   help='Space-separated queries in KEY=VALUE format or JSON string. Use @{file} to load from a file')
+                   help='Query parameters in the URL/URI. Space-separated queries in KEY=VALUE format or JSON string. '
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/util/_params.py:24]: `URI` is only used for backward compatibility. 

```suggestion
                   help='Query parameters in the URL. Space-separated queries in KEY=VALUE format or JSON string. '
```

File: src/azure-cli/azure/cli/command_modules/util/_help.py [hunk 4]
```diff
@@ -31,6 +31,9 @@
   - name: Create a public IP address from body.json file
     text: >
         az rest --method put --url https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/publicIPAddresses/{publicIpAddressName}?api-version=2019-09-01 --body @body.json
+  - name: List the top three resources
+    text: >
+        az rest --method get --url https://management.azure.com/subscriptions/{subscriptionId}/resources?api-version=2019-07-01 --url-parameters $top=3
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/util/_help.py:36]:
```suggestion
  - name: List the top three resources (Bash)
    text: >
        az rest --method get --url https://management.azure.com/subscriptions/{subscriptionId}/resources?api-version=2019-07-01 --url-parameters \$top=3
```

File: doc/use_cli_effectively.md [hunk 5]
```diff
@@ -122,6 +122,8 @@ az rest --method PATCH
         --body '{"web":{"redirectUris":["https://myapp.com"]}}'
 ```
 
+When using `--url-parameters`/`--uri-parameters` for requests in the form of OData, please pay attention to handling the escape `$` in different environments: in `Bash`, escape `$` as `\$` and in `PowerShell`, escape `$` as `` `$``
```
Comment (jiasli) [inline doc/use_cli_effectively.md:125]: The word `URI` is only used for backward compatibility.

```suggestion
When using `--url-parameters` for requests in the form of OData, please make sure to escape `$` in different environments: in `Bash`, escape `$` as `\$` and in `PowerShell`, escape `$` as `` `$``
```

**Pull request merged**